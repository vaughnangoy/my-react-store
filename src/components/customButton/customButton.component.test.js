import React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import CustomButton from './customButton.component';

describe('CustomButton', () => {
    afterEach(cleanup);
    it('renders', () => {
        const { asFragment } = render(<CustomButton />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render the children', () => {
        const { getByTestId, getByText } = render(
            <CustomButton>
                <div data-testid="test">Test Child Component</div>
            </CustomButton>,
        );
        expect(getByTestId('test')).toBeInTheDocument();
        expect(getByText('Test Child Component')).toBeInTheDocument();
    });
});
