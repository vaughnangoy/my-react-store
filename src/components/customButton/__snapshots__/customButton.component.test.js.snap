// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton renders 1`] = `
<DocumentFragment>
  <button
    class=" custom-button"
  />
</DocumentFragment>
`;
